### PR TITLE
Add a new parameter -Depth to Split-Path command

### PR DESCRIPTION
**Pull request by kvprasoon**
This new parameter will allow users to mention the depth of the split operation.
This will reduce the usage of multiple Split-Path command to get the parent of parent or leaf of parent.

#Default value is 0

### CODE REVIEW

File: src/System.Management.Automation/engine/PathInterfaces.cs [hunk 1]
```diff
@@ -1122,15 +1122,15 @@ internal string Combine(string parent, string child, CmdletProviderContext conte
         /// <exception cref="ProviderInvocationException">
         /// If the provider threw an exception.
         /// </exception>
-        public string ParseParent(string path, string root)
+        public string ParseParent(string path, int depth, string root)
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/PathInterfaces.cs]: We shouldn't change a public API. Please revert the change and add new overload.

File: src/System.Management.Automation/engine/SessionStateNavigation.cs [hunk 2]
```diff
@@ -384,7 +386,7 @@ private string RemoveQualifier(string path, ProviderInfo provider, out string qu
                 "Caller should validate context before calling this method");
 
             CmdletProvider providerInstance = GetProviderInstance(provider);
-            return GetParentPath(providerInstance, path, root, context);
+            return GetParentPath(providerInstance,depth , path, root, context);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/SessionStateNavigation.cs]: Formatting.

File: src/System.Management.Automation/namespaces/NavigationProviderBase.cs [hunk 3]
```diff
@@ -492,7 +493,7 @@ protected string MakePath(string parent, string child, bool childIsLeaf)
         /// for the provider namespace. For example, the file system provider should look
         /// for the last "\" and return everything to the left of the "\".
         /// </remarks>
-        protected virtual string GetParentPath(string path, string root)
+        protected virtual string GetParentPath(string path,int depth, string root)
```
Comment (iSazonov) [inline src/System.Management.Automation/namespaces/NavigationProviderBase.cs]: Formatting.
And maybe add new overload too.

**Review by iSazonov — COMMENTED** [review]
@kvprasoon Please add tests.

File: src/System.Management.Automation/namespaces/NavigationProviderBase.cs (hunk 3 above)
Comment (kvprasoon) [reply src/System.Management.Automation/namespaces/NavigationProviderBase.cs]: Sorry, new to C#.
So I need to create a new overload for ParseParent() has to be created and need to format the invocation lines ?

Comment (iSazonov) [reply src/System.Management.Automation/namespaces/NavigationProviderBase.cs]: @kvprasoon Please see docs https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/member-overloading
> DO make only the longest overload virtual (if extensibility is required). Shorter overloads should simply call through to a longer overload. 

So `GetParentPath(string path, string root)` should call `virtual GetParentPath(string path,int depth, string root)`

File: src/System.Management.Automation/engine/PathInterfaces.cs (hunk 1 above)
Comment (TravisEz13) [reply src/System.Management.Automation/engine/PathInterfaces.cs]: Otherwise, this is a breaking change...

**Comment by iSazonov** [comment]
@kvprasoon  It seems you revert all changes.

**Comment by kvprasoon** [comment]
Yes, I've reverted and will be making changes as per the guidelines .

**Comment by iSazonov** [comment]
@kvprasoon If you're experiencing any problems feel free to ask!
